polkadot/node/network/availability-distribution/src/requester/mod.rs

```diff
@@ -162,9 +198,125 @@ impl Requester {
 			self.add_cores(ctx, runtime, leaf, leaf_session_index, cores).await?;
 		}
 
+		let groups = get_validator_groups(sender, new_head.hash).await?;
+		self.early_request_chunks(ctx, runtime, new_head, leaf_session_index, &groups).await;
```

Comment:
This feels very early to me. It would only fetch availability for candidates that:
- were not know by the block author at authorship time. 
- have been backed before the block import on the local node

If we add some delay, it would make it possible to fetch candidates more candidates, that have just been backed after the active leaf is processed. Otherwise, these candidates will be fetched on the slower path (on the next block import).